File: go/test_lib/base_ball.go
package test_lib

import (
	// "fmt"
	"strconv"
)

func CalcPoints(input []string) int {

	var out []int = []int{}
	deleted := 0

	for _, ch := range input {
		val, err := strconv.Atoi(string(ch))

		if err == nil {
			out = append(out, val)
		} else {
			if ch == "D" {
				crt := out[len(out) - 1]
				out = out[0:len(out) - 1]
				out = append(out, crt, crt * deleted)
			}
			if ch == "C" {
				deleted = out[len(out) - 1]
				out = out[0:len(out) - 1]
			}
			if ch == "+" {
				x := out[len(out)-2:len(out)]
				out = out[0:len(out) - 2]
				out = append(out, x[0], x[1], x[0] + x[1])
			}
		}
	}
	
	sum := 0
	for _, elem  := range out {
		sum += elem
	}

	return sum
}

File: rust/test_lib/src/base_ball.rs

pub fn calc_points(input: &[&str]) -> i32 
{
    let mut out: Vec<i32> = vec![];

    let mut deleted = 0;

    for ch in input {
        let val0 = ch.parse::<i32>();

        match val0 {
            Ok(val) => {
                out.push(val);
            },
            Err(_e) => {
                if *ch == "D" {
                    let crt = out.pop().unwrap();
                    out.push(crt);
                    out.push(crt * deleted);
                } else if *ch == "C" {
                    deleted = out.pop().unwrap();
                } else if *ch == "+"{
                    let x1 = out.pop().unwrap();
                    let x0 = out.pop().unwrap();
                    out.push(x0);
                    out.push(x1);
                    out.push(x0 + x1);
                }
            },
        }
    }

    let mut sum = 0;

    for elem in out {
        sum += elem;
    }

    sum
}
